sway/desktop/render.c

```diff
@@ -2018,6 +2213,44 @@ void output_render(struct sway_output *output, struct timespec *when,
 			&output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BOTTOM]);
 		render_layer_popups(output, damage,
 			&output->layers[ZWLR_LAYER_SHELL_V1_LAYER_TOP]);
+
+		// Render the fullscreen containers on top
+		if (has_fullscreen) {
+			struct sway_workspace *workspaces[2] = { other_ws, workspace };
+			for (int i = 0; i < 2; i++) {
+				struct sway_workspace *ws =  workspaces[i];
+				if (!ws) {
+					// Render a blank rect next to the fullscreen container if
+					// there's nothing to render
+					struct wlr_box mon_box = { 0, 0, output->width, output->height };
+					adjust_box_to_workspace_offset(&mon_box, false, output);
+					scale_box(&mon_box, output->wlr_output->scale);
+					render_rect(output, damage, &mon_box, clear_color);
+
+					// Render a shadow to separate the edge and the fullscreen
+					// container
+					if (!config_should_parameters_shadow()) {
+						continue;
+					}
+					struct wlr_box shadow_box = { 0, 0, output->width, output->height };
+					adjust_box_to_workspace_offset(&shadow_box, true, output);
+					scale_box(&shadow_box, output->wlr_output->scale);
+					// Render rect to fix minor pixel gaps between fullscreen
+					// container and shadow
+					render_rect(output, damage, &shadow_box, clear_color);
+					render_box_shadow(output, damage, &shadow_box,
+							config->shadow_color, config->shadow_blur_sigma, 0,
+							0, 0);
+					continue;
+				}
```

Comment:
What conditions could result in this block being hit?

Reply:
If there's no sibling workspace next to the fullscreen view, we render a blank rect with a shadow. This only applies when scrolling in the “wrong” direction when the first or the last workspace is in focus.

![image](https://github.com/WillPower3309/swayfx/assets/35975961/3ce5187d-c1f5-4490-81c8-f70e8a0b9d69)

Reply:
Made this a bit more clear :)

Reply:
> If there's no sibling workspace next to the fullscreen view, we render a blank rect with a shadow. This only applies when scrolling in the “wrong” direction when the first or the last workspace is in focus.
> 
> ![image](https://github.com/WillPower3309/swayfx/assets/35975961/3ce5187d-c1f5-4490-81c8-f70e8a0b9d69)
> 

Oh yes, because we have an animation when you're at the "end", true!